retrieval_service/app/app_test.py

```diff
@@ -783,3 +808,80 @@ def test_policies_search(m_datastore, app, method_name, params, mock_return, exp
     assert len(output) == params["top_k"]
     assert output == expected
     assert models.Policy.model_validate(output[0])
+
+
+@patch.object(id_token, "verify_oauth2_token")
+@patch.object(datastore, "create")
+def test_insert_ticket_missing_user_info(m_datastore, m_verify_oauth2_token, app):
+    m_datastore = AsyncMock()
+    m_verify_oauth2_token.side_effect = ValueError("invalid token")
+    with TestClient(app) as client:
+        response = client.post(
+            "/tickets/insert",
+            json={
+                "airline": "CY",
+                "flight_number": "888",
+                "departure_airport": "LAX",
+                "arrival_airport": "JFK",
+                "departure_time": "2024-01-01 08:08:08",
+                "arrival_time": "2024-01-01 08:08:08",
+            },
+        )
+        assert response.status_code == 422
+        assert response.json()["detail"] == [
+            {
+                "type": "missing",
+                "loc": ["query", "airline"],
+                "msg": "Field required",
+                "input": None,
+                "url": "https://errors.pydantic.dev/2.6/v/missing",
+            },
+            {
+                "type": "missing",
+                "loc": ["query", "flight_number"],
+                "msg": "Field required",
+                "input": None,
+                "url": "https://errors.pydantic.dev/2.6/v/missing",
+            },
+            {
+                "type": "missing",
+                "loc": ["query", "departure_airport"],
+                "msg": "Field required",
+                "input": None,
+                "url": "https://errors.pydantic.dev/2.6/v/missing",
+            },
+            {
+                "type": "missing",
+                "loc": ["query", "arrival_airport"],
+                "msg": "Field required",
+                "input": None,
+                "url": "https://errors.pydantic.dev/2.6/v/missing",
+            },
+            {
+                "type": "missing",
+                "loc": ["query", "departure_time"],
+                "msg": "Field required",
+                "input": None,
+                "url": "https://errors.pydantic.dev/2.6/v/missing",
+            },
+            {
+                "type": "missing",
+                "loc": ["query", "arrival_time"],
+                "msg": "Field required",
+                "input": None,
+                "url": "https://errors.pydantic.dev/2.6/v/missing",
+            },
+        ]
+
+
+@patch.object(id_token, "verify_oauth2_token")
+@patch.object(datastore, "create")
+def test_list_tickets_missing_user_info(m_datastore, m_verify_oauth2_token, app):
+    m_datastore = AsyncMock()
+    m_verify_oauth2_token.side_effect = ValueError("invalid token")
```

Comment:
In this particular case (ie. failing case), no, since it would anyway fail with the dummy value. However, since I parameterised these tests, and since they now also include the success case, we do need to mock this now with the parameter provided. Note that still for the failure case, I just return `None`, and I still don't mock the side effect, since that would fail as intended. So thanks for the suggestion! That definitely simplifies 🙂